agent/background.go

```diff
@@ -68,7 +68,7 @@ func (a *Agent) startIdleTimeoutWatch(ctx context.Context, tc *taskContext, time
 				close(timeout)
 				return
 			}
-			a.checkIn(ctx, tc, tc.currentCommand, nextTimeout, resetIdleTimeout)
+			go a.checkIn(ctx, tc, nextTimeout, resetIdleTimeout)
```

Comment:
this should probably be blocking?

Reply:
No, that's the bug (I think). This function listens (line 72) on a channel that a.checkIn sends on. If we block on 71 then there's a deadlock. We were using a non-blocking send in task.go (line 128 of the old file) to prevent the deadlock but at the cost of this bug.

Reply:
What if checkIn backs up? I think then we get a much harder to find bug, where the agent thinks it's checked in and all is well, but the server thinks its gone rogue.

Reply:
checkIn doesn't check in with the server. It resets the agent's own idle timeout. I renamed it for clarity.